doc/vim-go.txt

```diff
@@ -1236,6 +1237,14 @@ Maximum height for the GoDoc window created with |:GoDoc|. Default is 20. >
   let g:go_doc_max_height = 20
 <
 
+                                                              *'g:go_doc_url'*
+
+godoc server URL used when |:GoDocBrowser| is used. Change if you want to use
+a private internal service. Default is 'https://godoc.org'. 
```

Comment:
What instance are you using? Is there a different service that doesn't has the `/pkg` in it?

Reply:
Basically just a glorified `for` loop which calls `godoc -html > "$file".html`

Reply:
So that's not an official solution then? How do you think to resolve this? I assume your approach is very unique and custom to yourself, everyone else just hosts `godoc --http`. I'm pretty sure the `/pkg` approach covers 99% of the people who're going to use this. Is there something I'm missing? :)

Reply:
Well, the `godoc` part is official :-)

My point was that perhaps it's a good idea to clarify what kind of service this supports, exactly. I originally assumed it was [gddo](https://github.com/golang/gddo/), but looking at this again it's actually just `godoc -http`?

Reply:
Actually it *does* support `gddo`, but only on `godoc.org`, but not anywhere else :-)

Reply:
What do you think about using a template like below? That'll work for `gddo`, `godoc -http`, and any custom crack that people may have, and it's (arguably) more elegant, too.

```
diff --git i/autoload/go/doc.vim w/autoload/go/doc.vim
index 7c2f53d..5691387 100644
--- i/autoload/go/doc.vim
+++ w/autoload/go/doc.vim
@@ -29,21 +29,13 @@ function! go#doc#OpenBrowser(...) abort
     let name = out["name"]
     let decl = out["decl"]
 
-    let godoc_url = get(g:, 'go_doc_url', 'https://godoc.org')
-    if godoc_url isnot 'https://godoc.org'
-      " strip last '/' character if available
-      let last_char = strlen(godoc_url) - 1
-      if godoc_url[last_char] == '/'
-        let godoc_url = strpart(godoc_url, 0, last_char)
-      endif
-
-      " custom godoc installations expects it
-      let godoc_url .= "/pkg"
-    endif
+    let godoc_url = get(g:, 'go_doc_url', 'https://godoc.org/%%IMPORTPATH%%#%%NAME%%')
+    let godoc_url = substitute(godoc_url, '%%IMPORTPATH%%', import, 'g')
 
-    let godoc_url .= "/" . import
     if decl !~ "^package"
-      let godoc_url .= "#" . name
+      let godoc_url = substitute(godoc_url, '%%NAME%%', name, 'g')
+    else
+      let godoc_url = substitute(godoc_url, '%%NAME%%', '', 'g')
     endif
 
     echo godoc_url
```

Reply:
Totally optional btw, this PR is already a good improvement.

Reply:
I'm not sure how this is going to work? I'm passing a custom `godoc_url` myself  (which doesn't has `IMPORTPATH` or `NAME` in it. In this is case it doesn't even resolve and just opens the base URL without the package name. 

Also there no sign of `/pkg` here. You can test it by just running `godoc -http=:6060` on your localhost and set `g:go_doc_url` to `localhost:6060`. And then call `:GoDocBrowser` on `fmt.Println` (put cursor on the word Println). You'll see it'll fail.

As far as I know, `gddo` is the source code for `godoc.org`. It's not intended to run separately for private repos. However people still use it and to make it work for private repositories it requires a fork to change the authorization mechanism (see: https://github.com/golang/gddo/issues/526).  Unless there is a better story there or documentation I think it's fair not to include it

I think the current implementation is a good start. Later we can improve upon more user feedback :)